src/dataqueue/queue.cc

```diff
@@ -0,0 +1,1221 @@
+#include "queue.h"
+#include <async_wrap-inl.h>
+#include <base_object-inl.h>
+#include <env-inl.h>
+#include <memory_tracker-inl.h>
+#include <node.h>
+#include <node_bob-inl.h>
+#include <node_errors.h>
+#include <node_external_reference.h>
+#include <node_file-inl.h>
+#include <stream_base-inl.h>
+#include <util-inl.h>
+#include <uv.h>
+#include <v8.h>
+#include <algorithm>
+#include <deque>
+#include <initializer_list>
+#include <memory>
+#include <vector>
+
+namespace node {
+
+using v8::ArrayBufferView;
+using v8::BackingStore;
+using v8::FunctionCallbackInfo;
+using v8::FunctionTemplate;
+using v8::Int32;
+using v8::Isolate;
+using v8::Just;
+using v8::Local;
+using v8::Maybe;
+using v8::Nothing;
+using v8::Object;
+using v8::Uint32;
+using v8::Value;
+
+namespace {
+// ============================================================================
+class IdempotentDataQueueReader;
+class NonIdempotentDataQueueReader;
+
+class EntryBase : public DataQueue::Entry {
```

Comment:
It’s really confusing naming that `EntryBase` inherits from `Entry` and not the other way around

Reply:
`EntryBase` should really be `EntryImpl`